### ISSUE TITLE
CX: Agent stucks on selected goals that cannot be expanded

### ISSUE THREAD

**Issue by TarikViehmann** [carologistics/fawkes-robotino#369]
I encountered the following issue:
 - Robot 1 executes a `FILL-CAP` goal and starts picking up a `CC` from the shelf.
 - Meanwhile robot 2 formulates and selects the same goal
 - Robot 1 finishes the action and holds the `CC`,
 - Robot 2 gets the sync update that the `CC` is no longer at the shelf
 - This fact is needed by Robot 2 in order to expand the `FILL-CAP` plan
Result: Robot 2 is stuck with a goal that is selected.

So basically what happened here is that between selection and expansion of a goal the worldmodel changed and renders expansion infeasible.
We currently do not deal with this, an easy solution would be to simply `FAIL` selected goals if they are not `EXPANDED`.
